scripts/tests/test_singularity_cmd.bats

```diff
@@ -13,27 +13,96 @@
 #   bats test_singularity_cmd.bats
 #   bats .
 
+load test_helpers
+
+arg_test_img="$BATS_TEST_DIRNAME/arg-test.simg"
+topdir="$BATS_TEST_DIRNAME/../.."
+
+cd "$BATS_TEST_DIRNAME"
+git annex get "$arg_test_img"
+
+
 @test "verifying arguments passed to singularity_cmd Docker shim" {
-    # make sure that we have the singularity image
-    img="$BATS_TEST_DIRNAME/arg-test.simg"
-    cd "$BATS_TEST_DIRNAME"
-    git annex get "$img"
-    cd ../..
-    # make sure that we have our shim docker image so its pulling does not
-    # leak into output of scripts/singularity_cmd
-    docker pull mjtravers/singularity-shim:latest
-    
-    export REPRONIM_USE_DOCKER=1
-    run scripts/singularity_cmd \
-        exec "$img" /singularity "foo bar" blah 45.5 /dir "bar;" "foo&" '${foo}'
-    echo "> STATUS=$status" >&3
-    echo "> lines=${lines[@]}" >&3
-    [ "$status" -eq 0 ]
-    [ "${lines[0]}" = 'arg #1=<foo bar>' ]
-    [ "${lines[1]}" = 'arg #2=<blah>' ]
-    [ "${lines[2]}" = 'arg #3=<45.5>' ]
-    [ "${lines[3]}" = 'arg #4=</dir>' ]
-    [ "${lines[4]}" = 'arg #5=<bar;>' ]
-    [ "${lines[5]}" = 'arg #6=<foo&>' ]
-    [ "${lines[6]}" = 'arg #7=<${foo}>' ]
+	pull_singularity_shim
+
+	cd "$topdir"
+	export REPRONIM_USE_DOCKER=1
+	run scripts/singularity_cmd \
+		exec "$arg_test_img" /singularity "foo bar" blah 45.5 /dir "bar;" "foo&" '${foo}'
+
+	debug_run
+
+	assert_clean_exit
+	assert_equal "${lines[0]}"	'arg #1=<foo bar>'
+	assert_equal "${lines[1]}"	'arg #2=<blah>'
+	assert_equal "${lines[2]}"	'arg #3=<45.5>'
+	assert_equal "${lines[3]}"	'arg #4=</dir>'
+	assert_equal "${lines[4]}"	'arg #5=<bar;>'
+	assert_equal "${lines[5]}"	'arg #6=<foo&>'
+	assert_equal "${lines[6]}"	'arg #7=<${foo}>'
+}
+
+
+@test "verifying ability to singularity exec under /tmp subdir" {
+    skip_if_travis_osx "skipping Singularity directory test on Travis OSX"
+	check_subdir "$(_mktemp_dir_under /tmp)"
+}
+
+@test "verifying ability to singularity exec under /tmp subdir (explicit use of docker)" {
+    skip_if_travis_osx "skipping Singularity directory test on Travis OSX"
+	export REPRONIM_USE_DOCKER=1
+	check_subdir "$(_mktemp_dir_under /tmp)"
+}
+
+@test "verifying ability to singularity exec under $HOME subdir" {
+    skip_if_travis_osx "skipping Singularity directory test on Travis OSX"
+	check_subdir "$(_mktemp_dir_under $HOME)"
+}
+
+@test "verifying ability to singularity exec under $HOME subdir (explicit use of docker)" {
+    skip_if_travis_osx "skipping Singularity directory test on Travis OSX"
```

Comment:
@chaselgrove thanks! minor note -- tabs (not spaces) used for indentation in all these bash scripts/tests AFAIK. would be nice to stay consistent - I will push the fix